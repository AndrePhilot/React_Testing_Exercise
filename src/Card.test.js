import { render } from "@testing-library/react";
import Card from "./Card";
import TEST_IMAGES from "./_testCommon.js";

it("renders without crashing", () => {
    render(<Card caption={TEST_IMAGES[0].caption} src={TEST_IMAGES[0].src} currNum={TEST_IMAGES.indexOf(TEST_IMAGES[0]) + 1} totalNum={TEST_IMAGES.length}/>);
});

it("matches snapshot for first element", () => {
    const { asFragment } = render(<Card caption={TEST_IMAGES[0].caption} src={TEST_IMAGES[0].src} currNum={TEST_IMAGES.indexOf(TEST_IMAGES[0]) + 1} totalNum={TEST_IMAGES.length}/>);
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot for second element", () => {
    const { asFragment } = render(<Card caption={TEST_IMAGES[1].caption} src={TEST_IMAGES[1].src} currNum={TEST_IMAGES.indexOf(TEST_IMAGES[1]) + 1} totalNum={TEST_IMAGES.length}/>);
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot for third element", () => {
    const { asFragment } = render(<Card caption={TEST_IMAGES[2].caption} src={TEST_IMAGES[2].src} currNum={TEST_IMAGES.indexOf(TEST_IMAGES[2]) + 1} totalNum={TEST_IMAGES.length}/>);
    expect(asFragment()).toMatchSnapshot();
});