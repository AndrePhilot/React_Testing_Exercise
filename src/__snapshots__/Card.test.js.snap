// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for first element 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testing image 1
    </h4>
    <img
      alt="testing image 1"
      class="Card-image"
      src="test1.com"
    />
    <small
      class="Card-small"
    >
      Image 1 of 3.
    </small>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot for second element 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testing image 2
    </h4>
    <img
      alt="testing image 2"
      class="Card-image"
      src="test2.com"
    />
    <small
      class="Card-small"
    >
      Image 2 of 3.
    </small>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot for third element 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testing image 3
    </h4>
    <img
      alt="testing image 3"
      class="Card-image"
      src="test3.com"
    />
    <small
      class="Card-small"
    >
      Image 3 of 3.
    </small>
  </div>
</DocumentFragment>
`;
